core/views.py

```diff
@@ -73,6 +74,17 @@ def save_hackathon(request, hackathon_id):
     return redirect("hackathons")
 
 
+@login_required
+def unsave_hackathon(request: HttpRequest, hackathon_id):
+    if request.method == "POST":
+        user = request.user
+        hackathon = get_object_or_404(Hackathon, id=hackathon_id)
+        user.saved.remove(hackathon)
```

Comment:
This doesn't handle the case that this endpoint gets called twice (due to wacky internet) and returns a 500 as the hackathon is already unsaved.

Reply:
Not sure how to handle this one

Reply:
You can simply add a if statement 

Attached is an example 
```py
from myapp.models import Article, Publication

article = Article.objects.get(id=1)
publication = Publication.objects.get(id=2)

if publication in article.publications.all():
    article.publications.remove(publication)
```

Reply:
I've only tried something like this with my superuser and it seems to not work... I don't remember the exact message but I'll keep you posted

Reply:
I don't get what you mean about your superuser but yeah, keep us posted!

Reply:
Nevermind, everything worked okay, I was initially getting the list incorrectly, that's all